pandas/tests/extension/base/setitem.py

```diff
@@ -189,11 +189,9 @@ def test_setitem_scalar_key_sequence_raise(self, data):
 
     def test_setitem_preserves_views(self, data):
         # GH#28150 setitem shouldn't swap the underlying data
-        assert data[-1] != data[0]  # otherwise test would not be meaningful
```

Comment:
if we start with `data[0] == data[1]` then the assertions below aren't testing anything useful.  some of the data fixtures check for a condition similar to this one

Reply:
We require that they be unequal: https://github.com/pandas-dev/pandas/blob/45d8d77f27cf0dbc8cefe932f8fb64f6982b9527/pandas/tests/extension/conftest.py#L19